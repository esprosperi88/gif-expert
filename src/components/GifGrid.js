
import React, { useState } from 'react';
import PropsTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import  {useFetchGifs} from '../hooks/useFetchGifs';

export const GifGrid =( {Categoria}) =>{

    const {data:gifts,loading} = useFetchGifs(Categoria);

    return <>
            <h3 className="animate__animated animate__fadeIn" >{Categoria}</h3>    
            <div className="card-grid">   
                {
                gifts.map(item => (
                      
                    <GifGridItem 
                                key={ item.id } 
                                {...item} />
                            
                    ))
                }
            </div>
            </>

}

GifGrid.propTypes={

    Categoria: PropsTypes.string.isRequired

}


export default GifGrid;