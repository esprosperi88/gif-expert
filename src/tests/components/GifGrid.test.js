import React from "react";
import GifGrid from '../../components/GifGrid';
import {shallow}   from 'enzyme'


describe('Prueba component - GifGrid ',() => {


    test('GifGrid - snapshot',() => {
        const saludo = 'hola';
        const wrapper = shallow(<GifGrid
            key="dragon"  
            Categoria= "dragon" 
        />);

        expect (wrapper).toMatchSnapshot();
    });

});