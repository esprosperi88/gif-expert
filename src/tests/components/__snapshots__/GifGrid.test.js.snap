// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba component - GifGrid  GifGrid - snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    dragon
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
